ciao-controller/configuration.go

```diff
@@ -0,0 +1,265 @@
+// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/01org/ciao/ciao-controller/types"
+	"github.com/01org/ciao/payloads"
+	"github.com/01org/ciao/service"
+	"github.com/01org/ciao/ssntp"
+	"github.com/golang/glog"
+	"gopkg.in/yaml.v2"
+)
+
+func (c *controller) UpdateClusterConfig(ctx context.Context, newConf types.ConfigRequest, token string) error {
+	var pyld payloads.Configure
+	tenantID, err := service.GetTenantID(ctx)
+	if err != nil {
+		return errors.New("Unable to retrieve tenant ID")
+	}
+	if newConf.Element == "" {
+		return errors.New("empty Element")
+	}
+	if newConf.Value == "" {
+		return errors.New("empty Value")
+	}
+	if validConfigElement(newConf.Element) == false {
+		return errors.New("invalid Value")
+	}
+	if validConfigValue(newConf.Value, newConf.Element) == false {
+		return errors.New("invalid Element")
+	}
+	pyld, err = c.client.ssntpClient().ClusterConfiguration()
+	if err != nil {
+		return err
+	}
+	err = testNewConfig(newConf, token, tenantID, pyld)
+	if err != nil {
+		return err
+	}
+	err = updateConfPayload(newConf, &pyld)
+	if err != nil {
+		return err
+	}
+	newConfPyld, err := yaml.Marshal(pyld)
+	if err != nil {
+		return err
+	}
+	c.client.ssntpClient().SendCommand(ssntp.CONFIGURE, newConfPyld)
+	return nil
+}
+
+func (c *controller) ShowClusterConfig() (string, error) {
+	pyld, err := c.client.ssntpClient().ClusterConfiguration()
+	if err != nil {
+		return "", err
+	}
+	// no need to share sensitive information
+	pyld.Configure.Controller.IdentityPassword = "*****"
+	pyld.Configure.Controller.AdminPassword = "*****"
+	pyld.Configure.Controller.AdminSSHKey = "*****"
+	s, err := yaml.Marshal(pyld)
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("%s", string(s)), nil
+}
+
+func updateConfPayload(confReq types.ConfigRequest, pyld *payloads.Configure) error {
+	badValueErr := fmt.Errorf("Unable to set '%s' value for '%s'", confReq.Value, confReq.Element)
+	switch confReq.Element {
+	case "scheduler.storage_uri":
+		pyld.Configure.Scheduler.ConfigStorageURI = confReq.Value
+	case "controller.compute_port":
+		newVal, err := strconv.Atoi(confReq.Value)
+		if err != nil {
+			return badValueErr
+		}
+		pyld.Configure.Controller.ComputePort = newVal
+	case "launcher.disk_limit":
```

Comment:
Just a comment here.  These changes won't be taken into affect until the launchers restart.  Launcher doesn't handle the SSNTP configure command.